### PR TITLE
Use enum for openshift operator statuses

### PR DESCRIPTION
**Pull request by guillaumerose**

### CODE REVIEW

File: pkg/crc/cluster/clusteroperator.go [hunk 1]
```diff
@@ -102,33 +102,31 @@ func getStatus(ocConfig oc.Config, selector []string) (*Status, error) {
...
-			case "Progressing":
-				if con.Status == "True" {
+			case openshiftapi.OperatorProgressing:
+				if con.Status == openshiftapi.ConditionTrue {
 					logging.Debug(c.ObjectMeta.Name, " operator is still progressing, Reason: ", con.Reason)
 					cs.progressing = append(cs.progressing, c.ObjectMeta.Name)
 					cs.Progressing = true
 				}
-			case "Upgradeable":
+			case openshiftapi.OperatorUpgradeable:
```
Comment (gbraad) [inline pkg/crc/cluster/clusteroperator.go:123]: The correct spelling is `Upgradable`... Upgradeable is an accepted variant but looks bonkers.

Comment (gbraad) [reply pkg/crc/cluster/clusteroperator.go:123]: Ugh

File: pkg/crc/cluster/clusteroperator.go [hunk 2]
```diff
@@ -102,33 +102,31 @@ func getStatus(ocConfig oc.Config, selector []string) (*Status, error) {
...
 				continue
-			case "Disabled":
-				if con.Status == "True" {
+			case "Disabled": // non official status, used by insights and cluster baremetal operators
```
[the remark this answers is not in the document]
Comment (cfergeau) [reply pkg/crc/cluster/clusteroperator.go:125]: `OperatorDisabled` would be more consistent with the other constants.

[the remark this answers is not in the document]
Comment (gbraad) [reply pkg/crc/cluster/clusteroperator.go:121]: We can define an internal state, like the noop case below in our own codebase.

**Review by gbraad — CHANGES_REQUESTED** [review]
Prevent string comparison; introduce a const that can easily be refactored/find references when needed.

**Review by anjannath — APPROVED** [review]

**Comment by gbraad** [comment]
it seems the majority does not care about maintainability; merging with a grudge

**Pull request merged**